src/amqp-ts.ts

```diff
@@ -398,17 +398,12 @@ export class Message {
       })
         .catch((error) => {
           exports.log.debug("Publish error: " + error.message, { module: "amqp-ts", error });
-          var destinationName = destination._name;
           var connection = destination._connection;
           exports.log.debug("Try to rebuild connection, before Call.", { module: "amqp-ts" });
 
           return connection._rebuildAll(error).then(() => {
-            exports.log.debug("Retransmitting message.", { module: "amqp-ts" });
-            if (destination instanceof Queue) {
-              return connection._queues[destinationName].publish(this.content, this.properties);
-            } else {
-              return connection._exchanges[destinationName].publish(this.content, routingKey, this.properties);
-            }
+            exports.log.info("Connection rebuilt, NOT retransmitting message.", { module: "amqp-ts" });
```

Comment:
This solution is preventing duplication but is not providing us with the guarantee that a message will be sent even by SQS. Should we throw here instead and retry on these errors?

Reply:
Adjusted by throwing an error. These are caught by the producer and retried in SQS :) Simply resolving was not cutting it in the case where we successfully rebuild the connection and complete the flow before the sunco timeout triggers https://github.com/zendesk/sunco/pull/10820/files